preprocessing-scripts/crop_image_and_detect_aimpoint.py

```diff
@@ -0,0 +1,172 @@
+#!/usr/bin/env python
+import argparse
+import json
+import logging
+import tempfile
+from pathlib import Path
+
+import requests
+import torch
+
+import paint.util.paint_mappings as mappings
+from paint import PAINT_ROOT
+from paint.data.stac_client import StacClient
+from paint.preprocessing.focal_spot_extractor import detect_focal_spot
+from paint.preprocessing.target_cropper import crop_image_with_template_matching
+from paint.util import set_logger_config
+
+set_logger_config()
+logger = logging.getLogger(__name__)
+
+
+def load_model_from_url(url: str) -> torch.jit.ScriptModule:
+    """
+    Download the model checkpoint from the given URL and load it directly without storing it permanently.
+
+    Parameters
+    ----------
+    url : str
+        URL of the model checkpoint.
+
+    Returns
+    -------
+    torch.jit.ScriptModule
+        Loaded PyTorch model.
+
+    Raises
+    ------
+    Exception
+        If the model cannot be loaded or the download fails.
+    """
+    logger.info(f"Downloading checkpoint from {url}...")
+    response = requests.get(url, stream=True)
+    if response.status_code == 200:
+        logger.info("Checkpoint downloaded successfully. Loading the model...")
+        with tempfile.NamedTemporaryFile(delete=True) as temp_file:
+            for chunk in response.iter_content(chunk_size=8192):
+                temp_file.write(chunk)
+            temp_file.flush()
+            try:
+                model = torch.jit.load(temp_file.name, map_location="cpu")
+                logger.info("Model loaded successfully.")
+                return model
+            except Exception as e:
+                logger.error(f"Failed to load the model: {e}")
+                raise
+    else:
+        logger.error(
+            f"Failed to download the checkpoint. Status code: {response.status_code}"
+        )
+        response.raise_for_status()
+
+
+def main(args: argparse.Namespace) -> None:
+    """
+    Process calibration images and extract focal spots using UTIS.
+
+    This function downloads and processes heliostat calibration data before loading the UTIS model from a given URL.
+    It then applies UTIS to detect focal spots for specified heliostats and measurement IDs.
+    """
+    # Ensure the output directory is a Path object.
+    output_dir = Path(args.output_dir)
+
+    # Create STAC client and download heliostat data.
+    client = StacClient(output_dir=output_dir)
+    client.get_heliostat_data(
+        heliostats=args.heliostats,
+        collections=args.collections,
+        filtered_calibration_keys=args.filtered_calibration,
+    )
```

Comment:
Currently you are still downloading all the IDs for a heliostat, but then only applying the extraction to one ID - have I understood this correctly? I would prefer to only download the image for the associated image ID. 

This should be simple to implement in the StacClient - I can try and look at this in the coming days. If you want to try it yourself then you can also have a go and ask me if you have questions :)